python/pdo/test/contract.py

```diff
@@ -339,38 +380,39 @@ def UpdateTheContract(config, enclave, contract, contract_invoker_keys) :
                 logger.error('enclave failed to evaluation expression; %s', str(e))
                 ErrorShutdown()
 
-            # if this operation did not change state then there is nothing
-            # to send to the ledger or to save
-            if not update_response.state_changed :
-                continue
-
-            try :
-                if ledger_config is not None :
-                    logger.debug("sending to ledger")
-                    # note that we will wait for commit of the transaction before
-                    # continuing; this is not necessary in general (if there is
-                    # confidence the transaction will succeed) but is useful for
-                    # testing
-                    txnid = update_response.submit_update_transaction(
-                        ledger_config,
-                        wait=30,
-                        transaction_dependency_list = txn_dependencies)
-                    txn_dependencies = [ txnid ]
-                else :
-                    logger.debug('no ledger config; skipping state save')
-            except Exception as e :
-                logger.error('failed to save the new state; %s', str(e))
-                ErrorShutdown()
-
-            logger.debug('update state')
-            contract.set_state(update_response.raw_state)
-
+            # if this operation did not change state then there is nothing to commit
+            if update_response.state_changed :
+                # asynchronously submit the commit task: (a commit task replicates change-set and submits the corresponding transaction)
+                try:
+                    logger.info("asynchronously replicate change set and submit transaction in the background")
+                    commit_id = update_response.commit_asynchronously(ledger_config,
+                        wait=30, dependency_list_txnids=[], dependency_list_commit_ids=commit_dependencies, use_ledger=use_ledger)
+                    commit_dependencies = [commit_id] 
+                except ReplicationException as e:
+                    logger.exception('failed to asynchronously start replication and transaction submission:' + str(e))
+                    ErrorShutdown()
+
```

Comment:
we have MANY levels of checks for replication exception. i'm not sure what it even means at this level.